### PR TITLE
Implemented takes response

### PR DESCRIPTION
**Pull request by g4s8**
#5 - implemented takes response wrapper which encapsulates Artipie response and implements Takes response

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/http/issues/10#issuecomment-581900255))

@g4s8 This is what I know about this job in [CT2E6TK9B](https://www.0crat.com/p/CT2E6TK9B), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#10](https://github.com/artipie/http/issues/10) is [in scope](http://www.zerocracy.com/policy.html#14) for 2hr
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fhttp%2310%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
This pull request [#10](https://github.com/artipie/http/issues/10) is assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/8456843f-01c2-488e-9583-f1db5165fdca); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/tk/TkResponse.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        public void onNext(final Byte item) {
+            try {
+                this.out.write(item);
+            } catch (final IOException err) {
+                throw new UncheckedIOException(err);
+            }
+        }
+
+        @Override
+        public void onError(final Throwable err) {
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/http/tk/TkResponse.java]: @g4s8 seems like we should close `OutputStream` here, [no other methods are invoked after this one](https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Flow.Subscriber.html#onError-java.lang.Throwable-)

**Comment by g4s8** [comment]
@olenagerasimova thanks, please see: https://github.com/artipie/http/pull/10/commits/63be0ff03b90c9b0b578367a7c76731b8843916b

**Comment by olenagerasimova** [comment]
@g4s8 I think we are good to merge here and I've also noticed that I have permissions to merge, is it ok?

**Comment by g4s8** [comment]
@olenagerasimova thanks. I'm closing this ticket, because the design of interfaces was changed.

**Comment by 0crat** [comment]
Job was finished in 40 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)

**Pull request closed without merging**